### ISSUE TITLE
Provide advice for dealing with the diaspora

### ISSUE THREAD

**Issue by richierocks** [hadley/devtools#1671]
I'd like some guidance on how the developers expect people to use *devtools* after the decomposition (into *remotes*, *usethis*, etc.) is complete. There are a few possible options.

### 1. Everyone still uses *devtools*

The business-as-usual option is that end users continue to use *devtools* just as before, and the decomposition into smaller packages is for the benefit of *devtools* developers only.

In this case, it would be useful to know how this is implemented. Would *devtools* attach the underlying packages (like *tidyverse* does)? Would it import and reexport everything (like *assertive* does)? Would it contain higher-level wrapper functions (like *rmarkdown* wraps *knitr*)?

### 2. Nobody uses *devtools*

The other extreme is that *devtools* is deprecated and everyone has to use the new packages for all tasks.

If this is the case, what's the timeline for deprecating the code? Are all the other packages ready for real-world use? That is, is it possible to completely ignore *devtools* yet?

### 3. *devtools* is used for some tasks

The in-between state is that devtools retains some functionality of its own, but other bits are farmed out exclusively to other packages.

In this case it would be useful to know which bits of functionality will stay in devtools and which bits will be fully outsourced.

----

The Diaspora section of the README states

> Generally, you should not need to worry about these different packages, because devtools installs them all automatically. You will need to care, however, if you're filing a bug because reporting it at the correct place will lead to a speedier resolution.

This suggests that (at least in the near future), it is expected that people are supposed to use *devtools*.

----

Contradicting this is that 

1. The `with_*` functions were deprecated and then removed once the *withr* was created.

2. The [`infrastructure.R`](https://github.com/hadley/devtools/blob/dcf77730da87ef7bdf45a5bd143c36c8e7490f11/R/infrastructure.R) file shows that the `use_*` functions are deprecated (and presumably will be removed shortly).

----

For context, DataCamp is making a course on creating R packages that features *devtools* quite heavily, and we need some guidance on what to tell students. Since video content is tricky to change, it would be helpful to know the long term plan.

**Comment by jimhester** [comment]
The plan is for devtools to be a meta package similar to tidyverse, so package authors will continue to use devtools to develop packages. However package authors should _not_ need to use devtools as a package dependency in their own packages. Rather they should depend on one of the smaller package that implements the specific functionality they need.

The exact mechanism of how the functions will be exposed is somewhat of an implementation detail, but will probably be a combination of just putting the relevant packages in Depends (attaching the packages) and in some cases importing with a wrapper.

The `with_*` functions were somewhat of a special case as they were primarily used in devtools internally rather than for users / package authors and were really orthogonal to devtools' purpose.

The infrastructure functions have all been moved to usethis. The functions will exist in devtools for a little while with deprecation messages, eventually they will be removed and usethis will be put in the Depends for devtools, so will be automatically attached when devtools is attached.

The bottom line is devtools will continue to exist and should be what you recommend students use now and in the future for package development.

**Comment by richierocks** [comment]
@jimhester Thanks for the clarification!

Just to be super clear on this, when students are taking the course in 2020 and one of the videos states "to build your package, you need to call the build() function in the devtools package", am I* going to have to deal with a load of complaints from students saying "build() isn't in the devtools package anymore, it's in pkgbuild"?

* By "I" I mean the DataCamp support team in the future.

**Comment by jimhester** [comment]
I don't think there is any need to worry. The plan is for devtools to always be a meta-package that contains the major build verbs (`build` being one of them). Most people will never need to know precisely which package the functions are coming from, the only time this could be important is when deciding which package to depend on.
